components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManager.java

```diff
@@ -66,6 +65,9 @@ private IdentitySCIMManager() throws CharonException {
      */
     public static IdentitySCIMManager getInstance() throws CharonException {
 
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            return new IdentitySCIMManager();
+        }
```

Comment:
Any reason to change this method?

Reply:
IdentitySCIMManager object is instantiated during server startup and set the endpoint URLs in AbstractResourceManager with the super tenant URLs at [1]. AbstractResourceManager is accessed by charon to get the endpoint URLs. So for tenants as well the endpoint URLs will the super tenant URLs. 
[1] https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/263/files/42a6b93ef0c49de05581e2fe2dc57dd4a3820aea#diff-609fb5ae29747e7d7564f70cbaa60e44R105

Reply:
this will instantiate a new instance of IdentitySCIMManager, everytime getInstance() is called. Better to have a instance variable and instantiate only if that variable is null.

Reply:
We should instantiate IdentitySCIMManager every time if we want to have tenant qualified URLs. 
Anyways we will be doing this only if tenant context is enabled. otherwise, it will be instantiated only once and the same instance will be used for the follow-up calls.

Reply:
We should not re-instantiate. Rather we should fix getters or add getters if not available to build url based on context